### DISCUSSION TITLE
Maps

### DISCUSSION THREAD

**Discussion by zohaibarsalan** [shadcn-ui/ui#4882] [Ideas]
Well now that we have charts in Shadcn the next big feature could be Maps. Many map libraries are available, but they aren't styled like Shadcn and don't fit well with the design consistency of my app because everything is Shadcn. 

It would be cool if the developers took a map library and designed and customized it similarly to the rest of the Shadcn design consistency.

**Comment by osmargm1202** [comment]
its hard to mantain same UI these days.

**Comment by iloveitaly** [comment]
Is there any good solution to a mapping component out there?

**Comment by nicolasperez19** [comment]
[Mapbox supports custom styles](https://www.mapbox.com/gallery)

**Comment by tonghohin** [comment]
Just built a map component based on shadcn/ui and Leaflet: https://shadcn-map.vercel.app/

**Reply by sakilanwar12** [reply]
#### Yes, I see the Shadcn Map. It's really good and I will use it .

But there is a bug in this project. When I navigate the layer-controls page, I see the issue.
I hope you will fix it.

<img width="1730" height="889" alt="image" src="https://github.com/user-attachments/assets/e3e788b9-f33a-4689-ad01-eaec64daefe5" />

**Reply by tonghohin** [reply]
Thanks for checking it out! I tried the page on a few different browsers but couldn’t reproduce the bug. I’m happy to fix it once I can see it happening. If you don’t mind, feel free to open an issue in the repo: https://github.com/tonghohin/shadcn-map

**Comment by AnmolSaini16** [comment]
You can check this out, custom map build with Next.js, Mapbox, shadcn ui, and Tailwind  - https://github.com/AnmolSaini16/next-maps
